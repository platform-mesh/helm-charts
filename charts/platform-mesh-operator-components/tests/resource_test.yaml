# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test the resource manifests
release:
  name: operator-components
  namespace: default
# values: 
#   - ../test-values.yaml
tests:
- it: override referencePath
  template: ocm-resources.yaml
  documentSelector:
    path: metadata.name
    value: virtual-workspaces
  set:
    componentVersion:
      semver: 0.0.0
    ocm:
      interval: 3m
      referencePath:
       - name: platform-mesh
      component:
        create: true
        name: platform-mesh
      repo:
        create: true
        name: platform-mesh
    services:
      virtual-workspaces:
        enabled: true
        dependsOn:
          - name: istio-istiod
            namespace: default
        referencePath:
          - name: test
  asserts:
    - equal:
        path: spec.resource.byReference.referencePath[0].name
        value: test
    - lengthEqual:
        path: spec.resource.byReference.referencePath
        count: 1


- it: default referencePath
  template: ocm-resources.yaml
  documentSelector:
    path: metadata.name
    value: tracing
  set:
    componentVersion:
      semver: 0.0.0
    ocm:
      interval: 3m
      referencePath:
       - name: platform-mesh
      component:
        create: true
        name: platform-mesh
      repo:
        create: true
        name: platform-mesh
    services:
      tracing:
        enabled: true
  asserts:
    - equal:
        path: spec.resource.byReference.referencePath[0].name
        value: platform-mesh
    - equal:
        path: spec.resource.byReference.referencePath[1].name
        value: tracing
    - lengthEqual:
        path: spec.resource.byReference.referencePath
        count: 2
