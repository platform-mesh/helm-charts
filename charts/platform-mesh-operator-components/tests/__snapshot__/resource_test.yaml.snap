image enabled resource:
  1: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        artifact: image
        component: etcd-druid
        repo: oci
      name: etcd-druid-image
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - apiVersion: delivery.ocm.software/v1alpha1
          kind: Repository
          name: platform-mesh
          namespace: default
      resource:
        byReference:
          referencePath:
            - name: etcd-druid
          resource:
            name: image
  2: |
    apiVersion: delivery.ocm.software/v1alpha1
    kind: Resource
    metadata:
      labels:
        component: infra
        infra: "true"
      name: kcp-operator-image
    spec:
      componentRef:
        name: platform-mesh
      interval: 1m
      ocmConfig:
        - apiVersion: delivery.ocm.software/v1alpha1
          kind: Repository
          name: platform-mesh
          namespace: default
      resource:
        byReference:
          referencePath:
            - name: kcp-operator
          resource:
            name: image
